### ISSUE TITLE
Plans which requires replacement of cluster raises cluster exists error

### ISSUE THREAD

**Issue by Kieran-Bacon** [terraform-aws-modules/terraform-aws-eks#2061]
## Description

If a plan requires the replacement of an EKS cluster, terraform will fail during the application of the plan as the cluster already exists.

I would expect that the original cluster be completed destroyed before the new cluster is attempted to be created.

## Actual behavior

```
  # module.eks.aws_eks_cluster.this[0] must be replaced
+/- resource "aws_eks_cluster" "this" {
      ~ arn                       = "arn:aws:eks:eu-west-2:****:cluster/****-consumers" -> (known after apply)
      ~ certificate_authority     = [
          - {
              - data = "LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSUM1ekNDQWMrZ0F3SUJBZ0lCQURBTkJna3Foa2lHOXcwQkFRc0ZBREFWTVJNd0VRWURWUVFERXdwcmRXSmwKY201bGRHVnpNQjRYRFRJeU1ETXdOekV3TlRVeE5sb1hEVE15TURNd05ERXdOVFV4Tmxvd0ZURVRNQkVHQTFVRQpBeE1LYTNWaVpYSnVaWFJsY3pDQ0FTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBTDVyCjVDTkhlbUtFZmd1ZDUzbmJIblBqMWNlTE5EQTk2NmVHd0dISTlKMmdnY3lBVzN5WWFvR2E1OFJuTGliQThHQVgKWHZCRldLU2JwRFpaRWxQdnlRNFE1RFVHclQzOUt4eTZEaS9zQ2VrQkJKajVzN0dUVTc2TE53VjFLWHVqNndwNAp2N2FxSlhVUlpRdit3U1dXREpjc1ZITnFrQVp3MER6VHhSaWppaDJFZ0h1QzBPb2hVNmg5QUwyRnVMR0l2MDRGCm9...
```

```
│ Error: error creating EKS Cluster (***-consumers): ResourceInUseException: Cluster already exists with name: ***-consumers
│ {
│   RespMetadata: {
│     StatusCode: 409,
│     RequestID: "4588f50d-b4d5-4f02-b953-75f93298645b"
│   },
│   ClusterName: "***-consumers",
│   Message_: "Cluster already exists with name: ***-consumers"
│ }
│ 
│   with module.eks.aws_eks_cluster.this[0],
│   on .terraform\modules\eks\main.tf line 11, in resource "aws_eks_cluster" "this":
│   11: resource "aws_eks_cluster" "this" {
```

## Versions

Terraform v1.1.9
on windows_386
+ provider registry.terraform.io/hashicorp/aws v4.13.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.2.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.11.0
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/template v2.2.0
+ provider registry.terraform.io/hashicorp/tls v3.3.0

**Comment by bryantbiggs** [comment]
This is not something we will control in the module - users can either run `terraform destroy` or rename the cluster

**Comment by Kieran-Bacon** [comment]
Hi @bryantbiggs 

I would imagine that many users (myself included) would not be able to run `terraform destroy` since there may be already established and unaffected resources that do not need to be destroyed and need to remain up, in the same terraform plan.

Furthermore, since the cluster name is used as the cluster identifier, things that record the name, or tags that reference it would be broken if I had to change the name every time there was an issue requiring a replacement.

At the end of the day - If I have made changes that require a replacement of the cluster, that is what I will end up doing. I really believe that this module should govern that. All of the resources linked to the cluster is known here, if I am to manage the tearing down and replacement of the cluster, I can easily do that in-correctly, it's laborious and resources can be left dangling.

If you are trying to avoid accidental destruction of a cluster in general then the use of the lifecycle `prevent_destroy` seems much appropriate. This would force people to explicitly say they are happy to destroy/replace the cluster.

In its current state replacing a cluster is a pain, and you do infact already destroy some but not all of the module resources

```
module.eks.aws_iam_openid_connect_provider.oidc_provider[0]: Destroying... [id=***C12F0D]
module.eks.aws_iam_openid_connect_provider.oidc_provider[0]: Destruction complete after 0s
...
```
